shared/kubernetes/podWatchHelper.go

```diff
@@ -0,0 +1,56 @@
+package kubernetes
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+type PodWatchHelper struct {
+	kubernetesProvider *Provider
+	NameRegexFilter    *regexp.Regexp
+}
+
+func NewPodWatchHelper(kubernetesProvider *Provider, NameRegexFilter *regexp.Regexp) *PodWatchHelper {
+	return &PodWatchHelper{
+		kubernetesProvider: kubernetesProvider,
+		NameRegexFilter: NameRegexFilter,
+	}
+}
+
+// Implements the EventFilterer Interface
+func (pwh *PodWatchHelper) Filter(e *watch.Event) (bool, error) {
+	pod, err := pwh.GetPodFromEvent(e)
+	if err != nil {
+		return false, nil
+	}
+
+	if !pwh.NameRegexFilter.MatchString(pod.Name) {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// Implements the WatchCreator Interface
+func (pwh *PodWatchHelper) NewWatcher(ctx context.Context, namespace string) (watch.Interface, error) {
+	watcher, err := pwh.kubernetesProvider.clientSet.CoreV1().Pods(namespace).Watch(ctx, metav1.ListOptions{Watch: true})
+	if err != nil {
+		return nil, err
+	}
+
+	return watcher, nil
+}
+
+func (pwh *PodWatchHelper) GetPodFromEvent(e *watch.Event) (*corev1.Pod, error) {
```

Comment:
why does it need to be executed on "PodWatchHelper"?

Reply:
I added the WatchEvent type which wraps apimachinery's Event and handles conversions.
https://github.com/up9inc/mizu/pull/470/commits/efb49f8799eae720943baf4a88117cbe56961bfe

Reply:
Thinking of changing `ToPod() *corev1.Pod, bool` to `AsPod(pod *corev1.Pod) bool`, like `errors.As`, so that it returns true/false and not an error. I think it is more "go". wdyt?